src/main.rs

```diff
@@ -149,7 +146,7 @@ async fn start(
 	let notifications_span = info_span!(parent: None, "cosmic-notifications");
 	let panel_span = info_span!(parent: None, "cosmic-panel");
 
-	let mut guard = notif_key.lock().unwrap();
+	let mut guard = notif_key.lock().await;
```

Comment:
```
warning: this `MutexGuard` is held across an `await` point
...
    = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
...
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
    = note: this warning originates in the macro `$crate::select` which comes from the expansion of the macro `tokio::select` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
That's a good catch. I haven't seen this lint before.